src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -168,6 +168,19 @@ jlong CgroupV1Subsystem::memory_and_swap_limit_in_bytes() {
   return memory_swap;
 }
 
+jlong CgroupV1Subsystem::memory_and_swap_usage_in_bytes() {
+  // isSwapEnabled - maybe compute once and save?
+  GET_CONTAINER_INFO(julong, _memory->controller(), "/memory.memsw.limit_in_bytes", "Memory and Swap Limit is: ", JULONG_FORMAT, JULONG_FORMAT, memsw_bytes);
+  GET_CONTAINER_INFO(julong, _memory->controller(), "/memory.swappiness", "Swappiness is: ", JULONG_FORMAT, JULONG_FORMAT, swappiness);
```

Comment:
Why are you reading the memory and swap limit by means of the low level files here? I think what you want to know is whether or not swap is enabled. Note that the `GET_CONTAINER_INFO` macro returns a `julong` (sometimes a negative value cast to `julong`). So you'd need to account for that.

I suggest something like this:

```
   jlong memory_sw_limit = memory_and_swap_limit_in_bytes();
   jlong memory_limit = CgroupSubsystem::memory_limit_in_bytes();
   if (memory_sw_limit > 0 && memory_limit > 0) {
       jlong delta_swap = memory_sw_limit - memory_limit;
       if (delta_swap > 0) {
                  GET_CONTAINER_INFO(julong, _memory->controller(), "/memory.memsw.usage_in_bytes",
                     "mem swap usage is: ", JULONG_FORMAT, JULONG_FORMAT, memory_swap_usage);
                return (jlong)memory_swap_usage;
       } else {
          // no swap
          return 0;
       }
       return memory_usage_in_bytes();
   }
```

Reply:
Thanks for the suggestion, I adjusted the coding .  Should I add some UL logging in case of negative values returned?

Reply:
I don't think that's needed.